sigstore/_internal/rekor/_client.py

```diff
@@ -2,8 +2,83 @@
 Client implementation for interacting with Rekor.
 """
 
+import json
+from urllib.parse import urljoin
+
+import requests
+
+DEFAULT_REKOR_URL = "https://rekor.sigstore.dev/api/v1/"
+
+
+class Endpoint:
+    def __init__(self, url, session):
+        self.url = url
+        self.session = session
+
+    def get(self):
```

Comment:
Maybe inherit from `abc.ABC` and mark these as `@abstractmethod`? I can push a commit if you're happy for me to do that.

Reply:
Please do!

Reply:
I don't think I have write permissions to your repo, could you add me? I have some commits for this + type hints and a few runtime checks.

Reply:
Added!